props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,720 @@
+        return;
+    }
+  }
+
+  bool IsEmpty() const {
+    return max_shots_ != -1 && shots_fired_ >= max_shots_;
+  }
+
+void Fire() {
+  if (IsEmpty() && blaster_mode != MODE_AUTO) {
+    SaberBase::DoEffect(EFFECT_EMPTY, 0);
+    return;
+  }
+#ifdef ENABLE_MOTION
+#ifdef BLASTER_JAM_PERCENTAGE
+  // If we're already jammed then we don't need to recheck. If we're not jammed then check if we just jammed.
+  is_jammed_ = is_jammed_ ? true : CheckJam(BLASTER_JAM_PERCENTAGE);
+  if (is_jammed_) {
+    SaberBase::DoEffect(EFFECT_JAM, 0);
+    return;
+  }
+#endif
+#endif
+  if (blaster_mode == MODE_STUN) {
+    SaberBase::DoEffect(EFFECT_STUN, 0);
+    shots_fired_++;
+    PVLOG_NORMAL << "******** STUN - Remaining shots = " << GetBulletCount() << "\n";
+  } else if (blaster_mode == MODE_KILL) {
+    SFX_blast.Select(-1);
+    SaberBase::DoEffect(EFFECT_FIRE, 0);
+    shots_fired_++;
+    PVLOG_NORMAL << "******** FIRE - Remaining shots = " << GetBulletCount() << "\n";
+  } else {
+    // Set up the autofire pairing if the font suits it.
+    if (!IsEmpty()) {
+      SelectAutoFirePair();
+      auto_firing_ = true;
+      SaberBase::SetLockup(LOCKUP_AUTOFIRE);
+      SaberBase::DoBeginLockup();
+    }
+  }
+}
+
+void Loop() override {
+  PropBase::Loop();
+  PollNextAction();
+  sound_library_.Poll(wav_player);
+  if (wav_player && !wav_player->isPlaying()) wav_player.Free();
+
+  if (blaster_mode == MODE_AUTO) {
+    // First time through, let bgnauto finish first if we have one
+    if (SFX_bgnauto && GetWavPlayerPlaying(&SFX_bgnauto)) return;
+
+    // No bgnauto or bgnauto has finished playing, so initialize and start auto-firing
+    if (auto_firing_ && !auto_player_) {
+      auto_player_ = GetWavPlayerPlaying(&SFX_auto);
+      shots_fired_++;
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+    // use auto.wav's length to count shots
+    if (auto_firing_ && auto_player_ && (millis() - auto_time_) > auto_sound_length) {
+      auto_sound_length = auto_player_->length() * 1000;
+      shots_fired_++;
+      if (IsEmpty()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_firing_ = false;
+        auto_player_.Free();
+        auto_time_ = millis(); // Reset for the first empty sound
+        return;
+      }
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+
+    // This will mimic the hammer running on no bullets at the same rate it was firing.
+    // if there are different length auto.wavs, the last played's length is used.
+    if (!auto_firing_ && IsEmpty() && trigger_is_pressed_ && (millis() - auto_time_) > auto_sound_length) {
+      SaberBase::DoEffect(EFFECT_EMPTY, 0);
+      auto_time_ = millis(); // Reset for the next empty sound
+    }
+  }
+}
+
+  void Reload() override {
```

Comment:
The only difference seems to be that you set is_jammed_ to false.
Not sure if we should just do that in Blaster.
If not, the function should probably call the base, like:

void Reload() override {
   Blaster::Reload();
   is_jammed_ = false;
}

Reply:
done